pkg/routing/localrouter.go

```diff
@@ -83,85 +91,117 @@ func (r *LocalRouter) ListNodes() ([]*livekit.Node, error) {
 }
 
 func (r *LocalRouter) StartParticipantSignal(ctx context.Context, roomName livekit.RoomName, pi ParticipantInit) (connectionID livekit.ConnectionID, reqSink MessageSink, resSource MessageSource, err error) {
+	return r.StartParticipantSignalWithNodeID(ctx, roomName, pi, livekit.NodeID(r.currentNode.Id))
+}
+
+func (r *LocalRouter) StartParticipantSignalWithNodeID(ctx context.Context, roomName livekit.RoomName, pi ParticipantInit, nodeID livekit.NodeID) (connectionID livekit.ConnectionID, reqSink MessageSink, resSource MessageSource, err error) {
 	prometheus.IncrementParticipantRtcInit(1)
-	// treat it as a new participant connecting
-	if r.onNewParticipant == nil {
-		err = ErrHandlerNotDefined
-		return
-	}
 
 	// create a new connection id
 	connectionID = livekit.ConnectionID(utils.NewGuid("CO_"))
-	// index channels by roomName | identity
-	key := participantKey(roomName, pi.Identity)
-	key = key + "|" + livekit.ParticipantKey(connectionID)
+	pKey := participantKeyLegacy(roomName, pi.Identity)
+	pKeyB62 := participantKey(roomName, pi.Identity)
```

Comment:
we don't need connectionID anymore because this was only used to associate messages from media -> signaling with the response channels. i'm all but certain we could deprecate legacy participant keys but reconnect would fail if a user began a session using a new signal node then tried reconnecting from a legacy signal node with the old key format... idk what failure modes we tolerate

Reply:
Got it.

On failure modes, would be good to have backwards compatibility and phased deprecation as projects using OSS and their own deploys should be able to upgrade without breaking compatibility. I need to spend more time thinking about the failure modes, but I am sure you have thought about compatibility. I am just stating the obvious 😄

Reply:
i feel like there's a maximum realistic amount of skew though. having a single deployment running 3 different versions seems like asking for bugs

Reply:
Can't agree more :-)